tools/snp.sh

```diff
@@ -1073,48 +1103,130 @@ generate_snp_expected_measurement() {
+    else
+      local expected_measurement=$(generate_snp_expected_measurement)
+      echo -e "\nExpected Measurement (sev-snp-measure): ${expected_measurement}"
+    fi
 
-  # Remove any special characters and print the value
-  snpguest_report_measurement=$(echo ${snpguest_report_measurement} | sed $'s/[^[:print:]\t]//g')
-  echo -e "Measurement from SNP Attestation Report: ${snpguest_report_measurement}\n"
+    # Parse the measurement out of the snp report
+    local snpguest_report_measurement=$(ssh_guest_command \
+        "./snpguest display report attestation-report.bin | \
+        tr '\n' ' ' | \
+        sed 's|.*Measurement:\(.*\)Host Data.*|\1|g' | \
+        sed 's| ||g'")
+
+    # Remove any special characters and print the value
+    snpguest_report_measurement=$(echo "$snpguest_report_measurement" | sed 's/[^[:print:]\t]//g')
+    echo -e "Measurement from SNP Attestation Report: ${snpguest_report_measurement}\n"
+
+    # Compare the expected measurement to the guest report measurement
+    if [ "${expected_measurement}" = "${snpguest_report_measurement}" ]; then
+      echo -e "The expected measurement matches the snp guest report measurement!"
+    else
+      >&2 echo -e "FAIL: measurements do not match"
+      return 1
+    fi
+}
 
-  # Compare the expected measurement to the guest report measurement
-  [[ "${expected_measurement}" == "${snpguest_report_measurement}" ]] \
-    && echo -e "The expected measurement matches the snp guest report measurement!" \
-    || { >&2 echo -e "FAIL: measurements do not match"; return 1; }
+setup_svsm_guest_attestation() {
+    ssh_guest_command <<'EOF'
+# Update and install necessary packages
+echo 'Updating package list...'
+sudo apt-get update
+
+echo 'Installing necessary packages...'
+sudo apt-get install -y git build-essential libtss2-dev tpm2-tools
+
+# Install Rust
+echo 'Installing Rust...'
+source "$HOME/.cargo/env" 2>/dev/null || true
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+source "$HOME/.cargo/env" 2>/dev/null
+
+# Create the 'tools' directory if it does not exist
+if [ ! -d "tools" ]; then
+    echo 'Directory tools does not exist. Creating it...'
+    mkdir tools
+else
+    echo 'Directory tools already exists.'
+fi
+
+# Change to 'tools' directory
+cd tools || { echo 'Failed to change directory to tools'; exit 1; }
+
+# Check if 'snpguest' directory exists and remove it if it does
+if [ -d "snpguest" ]; then
+    echo 'Directory snpguest exists. Removing it...'
+    rm -rf snpguest
+else
+    echo 'Directory snpguest does not exist.'
+fi
+
+# Clone the repository into the 'tools' directory
+git clone https://github.com/virtee/snpguest.git
+
+# Build the project inside the 'snpguest' directory
+cd snpguest || { echo 'Failed to change directory to snpguest'; exit 1; }
+source "$HOME/.cargo/env"
+cargo build --release
+
+# Copy the built binary to the home directory
+cp target/release/snpguest "$HOME/"
+EOF
 }
```

Comment:
Still need to remove this function. This is already done in another section of the code.

Reply:
My host is running Ubuntu 24.02, which has glibc version 2.39. If you build snpguest on this host, it won't run on an Ubuntu 22.04 guest VM because the guest uses glibc version 2.35. Therefore, I need to write a new function to build snpguest directly on the guest VM. (Error: ./snpguest: /lib/x86_64-linux-gnu/libc.so.6: version 'GLIBC_2.38' not found, required by ./snpguest). The existing function works if both the host and guest are running the same version of Ubuntu.

Reply:
The SVSM binary cannot be built on Ubuntu 22.04, which uses binutils version 2.35. SVSM requires binutils version 2.39 or higher (as documented in the Coconut-SVSM installation procedure), which is available on Ubuntu 24.04, the OS most commonly used by the community.

Reply:
> My host is running Ubuntu 24.02, which has glibc version 2.39. If you build snpguest on this host, it won't run on an Ubuntu 22.04 guest VM because the guest uses glibc version 2.35. Therefore, I need to write a new function to build snpguest directly on the guest VM. (Error: ./snpguest: /lib/x86_64-linux-gnu/libc.so.6: version 'GLIBC_2.38' not found, required by ./snpguest). The existing function works if both the host and guest are running the same version of Ubuntu.

If you are running into library compatibility issues, just build the musl release of the tool. This should fix the library issue.